### PR TITLE
Fix sink attention accuracy regression, add sink test and cleanup.

### PR DESCRIPTION
**Pull request by weireweire**
## 📌 Description

Update trtllm-gen cubin to fix accuracy regression about sink.

Integrate sink test to attention test as the old sink test didn't test and catch the fp8 accuracy issue before.

Clean up attention test code.

Attention test case has inflated a lot as param num raise. Reduce by using pairwise combination instead of Cartesian product.

## 🚀 Pull Request Checklist

Thank you for contributing to FlashInfer! Before we review your pull request, please make sure the following items are complete.

### ✅ Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` (or used your preferred method).
- [x] I have installed the hooks with `pre-commit install`.
- [ ] I have run the hooks manually with `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure about how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

## 🧪 Tests

- [x] Tests have been added or updated as needed.
- [ ] All tests are passing (`unittest`, etc.).

### CODE REVIEW

File: tests/test_trtllm_gen_attention.py [hunk 1]
```diff
@@ -161,7 +163,57 @@ def create_page_table(batch_size, seq_lens, page_size):
...
+            v_list.append(einops.rearrange(v_page, "h p d -> p h d"))
+    k_flat = torch.cat(k_list, dim=0)
+    v_flat = torch.cat(v_list, dim=0)
+    kv_indptr_tokens = torch.cat(
+        [
+            torch.tensor([0], dtype=torch.int32, device=device),
+            torch.cumsum(seq_lens.to(device), dim=0, dtype=torch.int32),
+        ]
+    )
+    return k_flat, v_flat, kv_indptr_tokens
```
[the remark this answers is not in the document]
Comment (weireweire) [reply tests/test_trtllm_gen_attention.py:204]: solved. thanks

**Comment by weireweire** [comment]
The added test can be a replacement of `test_attention_sink_blackwell.py` but I didn't remove it yet.
cc @IwakuraRein @joker-eph

File: flashinfer/artifacts.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def get_available_cubin_files(source, retries=3, delay=5, timeout=10):
 
 
 class ArtifactPath:
-    TRTLLM_GEN_FMHA: str = "538f8e38ace07f701f61e26b138b2b8c70ce9e8e/fmha/trtllm-gen/"
+    TRTLLM_GEN_FMHA: str = "7206d64e67f4c8949286246d6e2e07706af5d223/fmha/trtllm-gen/"
```
Comment (yzh119) [inline flashinfer/artifacts.py:72]: what's this change about?

Comment (weireweire) [reply flashinfer/artifacts.py:72]: fix sink attention accuracy regression，caused by adding the sink twice.

**Comment by yzh119** [comment]
@weireweire would you mind addressing the conflicts with main?

**Comment by elvischenv** [comment]
Hi @yzh119, could we bump a new version(maybe v0.4.0rc4) to include this PR? This could unblock the FP8 kv support for gpt-oss in vLLM. Thanks!
